.github/workflows/lint.yml

```diff
@@ -2,17 +2,55 @@ on:
   workflow_call:
     inputs:
       type:
-        description: "the expected lint entrypoint: a language name or either 'make' or 'just'"
-        default: "make"
+        description: "the expected lint entrypoint: make, just, or github"
+        default: "github"
+        required: false
+        type: string
+      language:
+        description: "the language to configure"
+        required: true
+        type: string
+      continue-on-error:
+        description: "don't mark the job as failed if a lint fails"
+        default: false
+        required: false
+        type: boolean
+      target:
+        description: "the make or just target to run for linting"
+        default: "lint"
         required: false
         type: string
```

Comment:
I think I'd like to limit flexibility here -- having `{make,just} lint` be the single valid "lint" target means that we provide a consistent development experience across our repos, and have a forcing function/incentive to update our existing repos.

Reply:
I added this so that I could run 'just lint-python' from the firebreak root so that it will only use just to lint python, and then use github to lint go. If I had a `just lint` it would make sense for it to lint both go and python, but I want native github actions linting for go. I ended up copying most of your python Makefile to each python dir so I can just have it call `make lint` with each dir specified now but its a lot of code duplication

Reply:
actually, what about setting variables for the target though? Using your makefile I usually run `make lint GROUP=lint`

Reply:
> I added this so that I could run 'just lint-python' from the firebreak root so that it will only use just to lint python, and then use github to lint go. If I had a just lint it would make sense for it to lint both go and python, but I want native github actions linting for go. I ended up copying most of your python Makefile to each python dir so I can just have it call make lint with each dir specified now but its a lot of code duplication

Yeah, this is one of those cases where a top-level `dev-requirements.txt` really shines...I should probably work out some guidance around that 😅 

> Using your makefile I usually run `make lint GROUP=lint`

Just making sure I understand: what does `GROUP=lint` do here? Or did you mean something like `GROUP=python`?

Reply:
Gotcha! Yeah, I think having that is a great idea -- IMO we should probably rename things like `BLIGHT_EXTRA` to `INSTALL_EXTRA` so that this workflow can reuse them generically. So if we go with `INSTALL_EXTRA` or `GROUP`, I'll update `blight` and the others to respect that 🙂

Reply:
(in which case IMO the setting should probably be `inputs.python-install-extra` or similar).

Reply:
`INSTALL_EXTRA` works for me!

Reply:
Sweet!